src/pages/main/hooks/useReferences.ts

```diff
@@ -1,22 +1,27 @@
-import { useRef } from "react";
+import { useCallback, useRef } from "react";
 
 import { editor } from "monaco-editor";
 
 export const useReferneces = () => {
   const monacoEditorRef = useRef<editor.IStandaloneCodeEditor | null>(null);
-  const setMonacoEditorRef = (
-    editorInstance: editor.IStandaloneCodeEditor | null,
-  ) => {
-    monacoEditorRef.current = editorInstance;
-  };
+  const setMonacoEditorRef = useCallback(
+    (editorInstance: editor.IStandaloneCodeEditor | null) => {
+      monacoEditorRef.current = editorInstance;
+    },
+    [],
+  );
   const iframeRefRef = useRef<HTMLIFrameElement | null>(null);
-  const setIframeRefRef = (iframeRef: HTMLIFrameElement | null) => {
+  const setIframeRefRef = useCallback((iframeRef: HTMLIFrameElement | null) => {
     iframeRefRef.current = iframeRef;
-  };
-  const isContentProgrammaticallyChanged = useRef<boolean>(false);
-  const setIsContentProgrammaticallyChanged = (value: boolean) => {
+  }, []);
+  const isContentProgrammaticallyChanged = useRef(false);
+  const setIsContentProgrammaticallyChanged = useCallback((value: boolean) => {
```

Comment:
What difference is wrapping it in useCallback is making. Just curious to know, as it is wrapped inside empty dependency

Reply:
so in the case, setIsContentProgrammaticallyChanged function would be redefiend when useReferences() is called.

Reply:
so in the case, setIsContentProgrammaticallyChanged would be redefiend when useReferences() is called.

Reply:
Ok cool, I will do more research on this. Memoization is not always good. I will check the benchmarks later.